internal/dyngo/operation.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package dyngo
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type (
+	// Option is the interface of operation options.
+	Option interface {
```

Comment:
I thought I saw a comment here about functional options. Let's use those and not interfaces please...